src/main/java/core/basesyntax/service/DataConverterImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> inputReport) {
+        return inputReport.stream()
+                .map(a -> a.trim())
+                .map(a -> a.split(","))
+                .filter(a -> isDataCorrect(a))
+                .map(el -> {
+                    FruitTransaction transaction = new FruitTransaction();
+                            transaction.setOperation(el[0]);
+                            transaction.setFruit(el[1]);
+                            transaction.setQuantity(Integer.valueOf(el[2]));

```

Comment:
There is no check for whether the conversion of 'el[2]' to Integer is safe. This could lead to a NumberFormatException if the input string cannot be parsed to an Integer.

Reply:
ok, done